ts/scene_io.ts

```diff
@@ -0,0 +1,55 @@
+import { readFile, writeFile } from 'fs';
+
+import { Scene } from './proto/scene';
+
+export type SceneIO = {
+
+    decode: (data: string | Buffer) => Scene,
+    read: (path: string) => Promise<Scene>,
+
+    encode: (scene: Scene) => Uint8Array,
+    save: (scene: Scene, path: string) => Promise<void>,
+    encodeAsJson: (scene: Scene) => string,
+    saveAsJson: (scene: Scene, path: string) => Promise<void>,
+};
+
+const decode = (data: Buffer | string) => {
+    try {
+        return Scene.fromJSON(JSON.parse(data.toString()));
+    } catch (e) {
+        if (Buffer.isBuffer(data)) {
+            return Scene.decode(data);
+        } else {
+            throw Error('Failed to decode as scene format: input is string, but not a valid json');
```

Comment:
Maybe it would be a good idea to show the JSON parse error details too, like which token in json invoked an error.

Reply:
Added `e` to message. Hope that contains some useful information :)